python/packages/azurefunctions/agent_framework_azurefunctions/_durable_agent_state.py

```diff
@@ -0,0 +1,1242 @@
+
+        # Restore the conversation history - deserialize entries from dicts to objects
+        history_data = data_dict.get("conversationHistory", [])
+        deserialized_history: list[DurableAgentStateEntry] = []
+        for entry_dict in history_data:
+            if isinstance(entry_dict, dict):
+                # Deserialize based on $type discriminator
+                entry_type = entry_dict.get("$type") or entry_dict.get("json_type")
+                if entry_type == "response":
+                    deserialized_history.append(DurableAgentStateResponse.from_dict(entry_dict))
+                elif entry_type == "request":
+                    deserialized_history.append(DurableAgentStateRequest.from_dict(entry_dict))
+                else:
+                    deserialized_history.append(DurableAgentStateEntry.from_dict(entry_dict))
+            else:
+                # Already an object
+                deserialized_history.append(entry_dict)  # type: ignore
+
+        self.data.conversationHistory = deserialized_history
+        self.data.extensionData = data_dict.get("extensionData")
+
+    @property
+    def message_count(self) -> int:
+        """Get the count of conversation entries (requests + responses)."""
+        return len(self.data.conversationHistory)
+
+    def try_get_agent_response(self, correlation_id: str) -> dict[str, Any] | None:
+        """Try to get an agent response by correlation ID.
+
+        This method searches the conversation history for a response entry matching the given
+        correlation ID and returns a dictionary suitable for HTTP API responses.
+
+        Note: The returned dictionary includes computed properties (message_count) that are
+        NOT part of the persisted state schema. These are derived values included for backward
+        compatibility with the HTTP API response format and should not be considered part of
+        the durable state structure.
+
+        Args:
+            correlation_id: The correlation ID to search for
+
+        Returns:
+            Response data dict with 'content', 'message_count', and 'correlationId' if found,
+            None otherwise
+        """
+        # Search through conversation history for a response with this correlationId
+        for entry in self.data.conversationHistory:
+            if (
+                hasattr(entry, "correlation_id")
+                and entry.correlation_id == correlation_id
+                and isinstance(entry, DurableAgentStateResponse)
+            ):
+                # Found the entry, extract response data
+                # Get the text content from assistant messages only
+                content = ""
+                for message in entry.messages:
+                    if hasattr(message, "role") and message.role == "assistant" and hasattr(message, "contents"):
+                        # Extract text from message contents
+                        for content_item in message.contents:
```

Comment:
Not all contents have `text` property, some have `message` also (like the `DurableAgentStateErrorContent`). 

Suggestion: Why not make `text` a top level property of `DurableAgentStateContent` base class and populate it always for all ContentTypes, that way parsing the `text` becomes way more simpler. 

```py
class DurableAgentStateContent:
    
    @property
    def text(self) -> str | None:
        """Get the text representation of this content item.

        Returns:
            Text string if this content has text, None otherwise

        Raises:
            NotImplementedError: Must be implemented by subclasses
        """
        raise NotImplementedError